tests/test_main.py

```diff
@@ -0,0 +1,126 @@
+import pytest
+import datetime
+from google.cloud import bigquery
+from mig_dx_api import CreatedBy, Installation
+from uuid import uuid4
+from unittest import mock
+from unittest.mock import patch
+
+from run import (
+    get_target_installation,
+    get_schema,
+    get_source_data,
+)
+
+TEST_APP_ID = str(uuid4())
+TEST_PRIVATE_KEY = 'private_key_contents'
+TEST_CREATOR = CreatedBy(
+    user_id = 123,
+    display_name = 'Test User',
+    email = 'test@example.com')
+
+def test_get_target_installation_default():
+    installations = [
+        Installation(
+            movement_app_id = str(uuid4()),
+            installation_id = 1,
+            workspace_id = 100,
+            created_by = TEST_CREATOR,
+            date_created = datetime.datetime.now(),
+            name = 'Test Installation'
+        )
+    ]
+
+    target_installation = get_target_installation(installations)
+    
+    assert target_installation == installations[0]
+
+def test_get_target_installation_with_install_id_argument():
+    installations = [
+        Installation(
+            movement_app_id = str(uuid4()),
+            installation_id = 1,
+            workspace_id = 100,
+            created_by = TEST_CREATOR,
+            date_created = datetime.datetime.now(),
+            name = 'Test Installation'
+        ),
+        Installation(
+            movement_app_id = str(uuid4()),
+            installation_id = 31,
+            workspace_id = 100,
+            created_by = TEST_CREATOR,
+            date_created = datetime.datetime.now(),
+            name = 'Test Installation 2'
+        )
+    ]
+    target_installation = get_target_installation(installations, '31')
+    
+    assert target_installation == installations[1]
+
+def test_get_target_installation_no_installations():
+    with pytest.raises(Exception):
```

Comment:
Could be nice to validate that it fails specifically with "No valid installations found", but not a blocker.